src/Sentry/Laravel/Features/HttpClientIntegration.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Sentry\Laravel\Features;
+
+use GuzzleHttp\Psr7\Uri;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Http\Client\Events\ConnectionFailed;
+use Illuminate\Http\Client\Events\RequestSending;
+use Illuminate\Http\Client\Events\ResponseReceived;
+use Illuminate\Http\Client\Factory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Sentry\Breadcrumb;
+use Sentry\Laravel\Features\Concerns\TracksPushedScopesAndSpans;
+use Sentry\Laravel\Integration;
+use Sentry\SentrySdk;
+use Sentry\Tracing\SpanContext;
+use Sentry\Tracing\SpanStatus;
+use function Sentry\getBaggage;
+use function Sentry\getTraceparent;
+
+class HttpClientIntegration extends Feature
+{
+    use TracksPushedScopesAndSpans;
+
+    private const FEATURE_KEY = 'http_client_requests';
+
+    public function isApplicable(): bool
+    {
+        return $this->isTracingFeatureEnabled(self::FEATURE_KEY)
+            || $this->isBreadcrumbFeatureEnabled(self::FEATURE_KEY);
+    }
+
+    public function onBoot(Dispatcher $events, Factory $factory): void
+    {
+        if ($this->isTracingFeatureEnabled(self::FEATURE_KEY)) {
+            $events->listen(RequestSending::class, [$this, 'handleRequestSendingHandlerForTracing']);
+            $events->listen(ResponseReceived::class, [$this, 'handleResponseReceivedHandlerForTracing']);
+            $events->listen(ConnectionFailed::class, [$this, 'handleConnectionFailedHandlerForTracing']);
+
+            // The `globalRequestMiddleware` functionality was introduced in Laravel 10.14
+            if (method_exists($factory, 'globalRequestMiddleware')) {
+                $factory->globalRequestMiddleware([$this, 'attachTracingHeadersToRequest']);
+            }
```

Comment:
We have to move this out of this scope, as we always want to apply the headers.

Reply:
I figured it would be nice to be able to disable it, but I guess you already can using `trace_propagation_targets` of course so there is no need to gate it behind the tracing config 👍